pkg/monitoring/metrics/common/client/metrics.go

```diff
@@ -31,10 +31,12 @@ import (
 
 var resourceParsingRegexs []*regexp.Regexp
 
-func SetupMetrics() error {
+func RegisterRestConfigHooks() {
```

Comment:
IIUC `RegisterRestConfigHooks` should always be executed before `GetKubevirtClientConfig`.
If this is true, maybe it is worth it to add a comment to document this expected behavior

Reply:
added

Reply:
If the default is to add a hook, in the future releases can we make this part of the `GetKubevirtClientConfig` function?

For now this looks good

Reply:
@machadovilaca Hi, can you please elaborate on the dependency between the two in the commit msg?

Reply:
@alaypatel07 There might be use cases where you don't want metrics, e.g tests...

Reply:
@xpivarc Then I should express whether I want metrics or not via the code i.e. via function initialization. How would I remember this dependency?

Reply:
@enp0s3 that is possible. I believe this was not done as the code is following kubernetes client-go